src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -49,15 +48,23 @@ func (a *App) onBootstrapWorkerNode(s *state.State, encodedToken string) error {
 	if nodeIP == nil {
 		return fmt.Errorf("failed to parse node IP address %s", s.Address().Hostname())
 	}
-
 	// TODO(neoaggelos): figure out how to use the microcluster client instead
-	// create an HTTP client that ignores https
-	httpClient := &http.Client{
-		Transport: &http.Transport{
-			TLSClientConfig: &tls.Config{
-				InsecureSkipVerify: true,
-			},
-		},
+
+	// Get remote certificate from the cluster member
+	cert, err := utils.GetRemoteCertificate(token.JoinAddresses[0])
+	if err != nil {
+		return fmt.Errorf("failed to get certificate of cluster member: %w", err)
+	}
+
+	// verify that the fingerprint of the certificate matches the fingerprint of the token
+	fingerprint := utils.CertFingerprint(cert)
+	if fingerprint != token.Fingerprint {
+		return fmt.Errorf("server authentication failed: join token fingerprint does not match that of the cluster member")
```

Comment:
let's make sure this error message helps people understand what is happening
```suggestion
		return fmt.Errorf("fingerprint from token (%q) does not match fingerprint of node %q (%q) (from node %v)", token.Fingerprint, fingerprint, token.JoinAddresses[0])
```